### ISSUE TITLE
Assert.Fail() does not publish failed test

### ISSUE THREAD

**Issue by MurrayLong** [nunit/nunit#3471]
I have a single test failing with Assert.Fail();   
But no failures are listed in the trx output, as below.
Passing a message argument to Assert.Fail() fixes the problem.

test.cs:
```
    [TestFixture]
    public class ExampleTest
    {
        [Test]
        public void This_test_should_fail()
        {
            Assert.Fail();
        }

        [Test]
        public void This_test_should_pass()
        {
            Assert.Pass();
        }
    }
```

```
>dotnet test --logger trx --results-directory .
...
Test Run Failed.
Total tests: 2
     Passed: 1
     Failed: 1
```

```
>cat *.trx
...
 </TestLists>
  <ResultSummary outcome="Completed">
    <Counters total="1" executed="1" passed="1" failed="0" error="0" timeout="0" aborted="0" inconclusive="0" passedButRunAborted="0" notRunnable="0" notExecuted="0" disconnected="0" warning="0" completed="0" inProgress="0" pending="0" />

```

**Comment by rprouse** [comment]
We'll need to repro this in the NUnit Console and determine if it is an NUnit issue, an adapter issue or a `dotnet test` issue.

**Comment by rprouse** [comment]
First pass, running this as a .NET 4.7.2 test in Visual Studio works fine, so it isn't NUnit or the NUnit Adapter.

![image](https://user-images.githubusercontent.com/493828/75506381-d1a8c600-59ab-11ea-9336-f5ee61c31a3b.png)

I'm also getting the same result with a .NET Core test assembly in Visual Studio.

`dotnet test` is also working;

```
dotnet test
...
Test Run Failed.
Total tests: 2
     Passed: 1
     Failed: 1
 Total time: 1.1454 Seconds
```

and I am getting the same results using the TRX logger,

```
dotnet test --logger trx --results-directory .
...
Test Run Failed.
Total tests: 2
     Passed: 1
     Failed: 1
 Total time: 1.1454 Seconds
```

Clearly something is different in your setup. I am using .NET Core 3.1. What about you?

How about the latest NUnit Adapter and Microsoft.NET.Test.Sdk, version 16.5.0?

**Comment by MurrayLong** [comment]
Microsoft.NET.Test.Sdk: 16.5.0
NUnit Adapter   3.16.1.0
NUnit: 3.12.0
TargetFramework: netcoreapp3.1

Did you check the contents of the .trx itself?

I see:
`  
<Counters total="1" executed="1" passed="1" failed="0" error="0" timeout="0" aborted="0" inconclusive="0" passedButRunAborted="0" notRunnable="0" notExecuted="0" disconnected="0" warning="0" completed="0" inProgress="0" pending="0" />
`

**Comment by rprouse** [comment]
@MurrayLong, you are correct, I am seeing the same in the TRX file. This may be an NUnit Adapter issue, or it could be a VSTest issue. I will be moving this issue to the adapter.

This is also being tracked in the VSTest repo at https://github.com/microsoft/vstest/issues/2319
